Zilon.Core/Zilon.Core.Tests/Persons/SurvivalDataTests.cs

```diff
@@ -11,6 +11,24 @@ namespace Zilon.Core.Tests.Persons
     [TestFixture]
     public class SurvivalDataTests
     {
+        /// <summary>
+        /// This test is to show the value variable is being range checked.
+        /// </summary>
+        [Test]
+        public void Update_TestStatValueItemSetter()
+        {
+            //ARRANGE
+            var survivalData = new SurvivalData();
+            var stat = survivalData.Stats.SingleOrDefault(x => x.Type == SurvivalStatType.Satiety);
+            //ACT
+            stat.Value = 0;
+            stat.Value.Should().Be(0);
+            stat.Value = stat.Range.Min - 2;
+            stat.Value.Should().Be(stat.Range.Min);
+            stat.Value = stat.Range.Max + 2;
+            stat.Value.Should().Be(stat.Range.Min);
+        }
+
```

Comment:
This should tests a SurvivalStat, not SurvivalData. In fact, after transferring the value check into the setter, SurvivalData does not participate in the adjustment of the value. Accordingly, the test itself must be in the class SurvivalStatTests.

I think it's better if one test checks single assertion. For this, NUnit has a cool tool for tests that only differ in data: TesCaseAttribute (https://github.com/nunit/docs/wiki/TestCase-Attribute) or TestCaseSourceAttribute (https://github.com/nunit/docs/wiki/TestCaseData).
In this case, the test should look something like this:
```c#
[TestFixture]
    public class SurvivalStatTests
    {
        /// <summary>
        /// This is the range being checked.
        /// </ summary>
        [TestCase (0, -2, 2, 1, ExpectedResult = 1)]
        [TestCase (0, -2, 2, -2, ExpectedResult = -2)]
        public int ValueSetter_ValueInsideRange_ValueHaventBeCorrected (int start, int min, int max, int newValue)
        {
            // ARRANGE
            var survivalStat = new SurvivalStat (start)
            {
                Range = new Range <int> (min, max)
            };



            // ACT
            survivalStat.Value = newValue;



            // ASSERT
            return survivalStat.Value;
        }

        /// <summary>
        /// This is the range being checked.
        /// </ summary>
        [TestCase (0, -2, 2, 3, ExpectedResult = 2)]
        [TestCase (0, -2, 2, -3, ExpectedResult = -2)]
        public int ValueSetter_ValueOutsideRange_ValueShiftedIntoRange (int start, int min, int max, int newValue)
        {
            // ARRANGE
            var survivalStat = new SurvivalStat (start)
            {
                Range = new Range <int> (min, max)
            };



            // ACT
            survivalStat.Value = newValue;



            // ASSERT
            return survivalStat.Value;
        }
    }
```

Reply:
But here, too, is not all right. Because checking the `Value` in the constructor will refer to `Range`, which is null.

You need to extend the constructor with the `min` and `max` parameters and before setting `Value` perform
```c#
Range = new Range <int> (min, max)
```
Then you can remove the setter for the `Range` property.

This will hurt other tests. Object initializer have been changed by using a constructor  :).